test/unit/decorators/grpc.decorator.unit.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback4-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+import {expect} from '@loopback/testlab';
+import {grpc} from '../../..';
+import {Application} from '@loopback/core';
+import {GrpcBindings} from '../../../src/keys';
+import {
+  GreeterInterface,
+  HelloRequest,
+  HelloReply,
+} from '../../protos/greeter.proto';
+import {Reflector} from '@loopback/context';
+
+describe('@rpc decorator', () => {
+  it('defines reflection metadata for rpc method', () => {
+    class Greeter implements GreeterInterface {
+      @grpc()
+      SayHello(request: HelloRequest): HelloReply {
+        return {message: `hello ${request.name}`};
+      }
+      Helper(): boolean {
+        return true;
+      }
+    }
+    let controllerMethods: string[] = [];
+    const flags: {[key: string]: boolean} = {};
+    const app: Application = givenApplication();
+    app.controller(Greeter);
+    const controller = app.find('controllers.Greeter').pop();
```

Comment:
Since this code is living in the file called `grpc.decorator.unit.ts`, it should contain a unit test that's testing the decorator in isolation, stubbing any dependencies as needed. You should not be using the full Application object! Unfortunately, we are behind on docs for extension developers (see https://github.com/strongloop/loopback-next/issues/579), but some relevant information can be found in these places:

 - http://loopback.io/doc/en/lb4/Testing-your-application.html#types-of-tests
 - http://loopback.io/doc/en/lb4/Testing-your-application.html#unit-testing

In this particular case, just scan `Greeter.prototype` directly.

```ts
let controllerMethods: string[] = [];
const flags: {[key: string]: boolean} = {};
const proto = Greeter.prototype;
// ...
```